doc/dev/mempool_accounting.rst

```diff
@@ -0,0 +1,51 @@
+What is a mempool?
+-----------------
+A memory pool is a method for accounting the consumption of memory of
+a set of containers.
+Memory pools are statically declared (see pool_index_t) Ultimately, which serves
+the purpose of identifying memory related checks in bluestore.
+
+Some common mempools that we can track
+--------------------------------------
+
+- bloom_filter: Ceph uses bloom filters to account for tracking objects already
+  in the cache pool, in order to determine which objects in the pool are being
+  accessed.
+- bluestore_alloc: It accounts for actual allocations for bitmap allocator or
+  stupid allocator.
+- bluestore_inline_bl: measures encoded length of an inline buffer.
+- bluestore_fsck: file system consistency check for bluestore metadata, helps in
+  tracking and debugging during OSD repairs.
+- bluestore_txc: accounts for committed transactions counter.
+- bluestore_writing_deferred: TBD
+- bluestore_writing: TBD
+- bluefs:  file-system-like interface that provides just enough functionality to
+  allow RocksDB to store its “files” and share the same raw device(s) with
+  BlueStore.
+- buffer_anon: stores arbitrary buffer data.
+- buffer_meta: all the metadata associated with buffer anon buffers.
+- bluestore_cache_data: mempool for writing and writing deferred.
+- bluestore_cache_onode: meta information of object node stored directed in bluestore cache.
+- bluestore_cache_meta: key under PREFIX_OBJ where we are stored.
+- bluestore_cache_other: right now account for:
+   * map_t
+   * coll_map
+   * csum_data
+- bluestore_cache_buffer: accounts for buffer cache shards.
+- bluestore_extent: a logical as well as physical extents, pointing to (some
+  portion of) a blob.
+- bluestore_blob : in-memory blob metadata and associated cached buffers.
+- bluestore_shared_blob: in memory shared blob state, stores reference of set of
+  collections it is part of (includes cached buffers).
+- bluefs_file_reader: accounts for bluefs file reader buffer.
+- bluefs_file_writer: accounts for bluefs file writer buffer.
+
+Check mempools usage
+~~~~~~~~~~~~~~~~~~~~
+
+Command to see bluestore memory allocation in these mempools::
+
+     $ ceph daemon osd.NNN dump_mempools
+
+
+.. see_more:: https://github.com/ceph/ceph/blob/fccbdc0905e3868666fbb10803bac6b73f687cb1/src/include/mempool.h#L159
```

Comment:
```
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/mempool_accounting.rst:2: WARNING: Title underline too short.

What is a mempool?
-----------------
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/mempool_accounting.rst:51: WARNING: Unknown directive type "see_more".

.. see_more:: https://github.com/ceph/ceph/blob/fccbdc0905e3868666fbb10803bac6b73f687cb1/src/include/mempool.h#L159
looking for now-outdated files... none found
```

https://jenkins.ceph.com/job/ceph-pr-docs/43361/consoleFull#-85362248744e9240e-b50a-4693-bac0-8a991bac86ac

I don't think we should reference the line number here, it is bound to change very easily.